tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -0,0 +1,215 @@
+  return &IOTNI;
+}
+
+/* Creates a Connection Info and assigns memory accordingly. */
+IotHttpsConnectionInfo_t * newConnectionInfo() {
+  IotHttpsConnectionInfo_t * pConnInfo = safeMalloc(sizeof(IotHttpsConnectionInfo_t));
+  if(pConnInfo) {
+    pConnInfo->pNetworkInterface = newNetworkInterface();
+    /* Function _createHttpsConnection checks if pConnInfo->addressLen is less than
+    IOT_HTTPS_MAX_HOST_NAME_LENGTH. The +1 is added to cover all cases. */
+    __CPROVER_assume(pConnInfo->addressLen >= 0 && pConnInfo->addressLen <= IOT_HTTPS_MAX_HOST_NAME_LENGTH + 1);
+    pConnInfo->pAddress = safeMalloc(pConnInfo->addressLen);
+    /* Function _createHttpsConnection checks if pConnInfo->alpnProtocolsLen is less than
+    IOT_HTTPS_MAX_ALPN_PROTOCOLS_LENGTH. The +1 is added to cover all cases. */
+    __CPROVER_assume(pConnInfo->alpnProtocolsLen >= 0 && pConnInfo->alpnProtocolsLen <= IOT_HTTPS_MAX_ALPN_PROTOCOLS_LENGTH + 1);
+    pConnInfo->pAlpnProtocols = safeMalloc(pConnInfo->alpnProtocolsLen);
+    pConnInfo->pCaCert = malloc(sizeof(uint32_t));
+    pConnInfo->pClientCert = malloc(sizeof(uint32_t));
+    pConnInfo->pPrivateKey = malloc(sizeof(uint32_t));
+    pConnInfo->userBuffer.pBuffer = safeMalloc(sizeof(_connHandle_t));
+  }
+  return pConnInfo;
+}
+
+/* Creates a Connection Handle and assigns memory accordingly. */
+IotHttpsConnectionHandle_t newIotConnectionHandle () {
+  IotHttpsConnectionHandle_t pConnectionHandle = safeMalloc(sizeof(_connHandle_t));
+  if(pConnectionHandle) {
+    pConnectionHandle->pNetworkConnection = safeMalloc(sizeof(uint32_t));
+    pConnectionHandle->pNetworkInterface = newNetworkInterface();
+    pConnectionHandle->reqQ.pPrevious = NULL;
+    pConnectionHandle->reqQ.pNext = NULL;
+    pConnectionHandle->respQ.pPrevious = NULL;
+    pConnectionHandle->respQ.pNext = NULL;
+    //pConnectionHandle->taskPoolJob = malloc(sizeof(struct _taskPoolJob));
+  }
+  return pConnectionHandle;
+}
+
+/* Creates a Response Handle and assigns memory accordingly. */
+IotHttpsResponseHandle_t newIotResponseHandle() {
+  IotHttpsResponseHandle_t pResponseHandle = safeMalloc(sizeof(_resHandle_t));
+  if(pResponseHandle) {
+    pResponseHandle->pBody = malloc(sizeof(uint32_t));
+    pResponseHandle->pHttpsConnection = newIotConnectionHandle();
+    pResponseHandle->pHttpsRequest = safeMalloc(sizeof(_reqHandle_t));
+    if(pResponseHandle->pHttpsRequest) {
+      pResponseHandle->pHttpsRequest->pHttpsResponse = pResponseHandle;
+    }
+    pResponseHandle->pHeaders = safeMalloc(sizeof(pResponseHandle) + sizeof(struct _httpResponse));
+    pResponseHandle->pHeadersCur = pResponseHandle->pHeaders;
+    pResponseHandle->httpParserInfo.readHeaderParser.data = pResponseHandle;
+    pResponseHandle->httpParserInfo.parseFunc = http_parser_execute;
+    pResponseHandle->pReadHeaderValue = safeMalloc(sizeof(uint32_t));
+    pResponseHandle->pHeadersEnd = safeMalloc(sizeof(pResponseHandle) + sizeof(_resHandle_t)); ;
```

Comment:
Extra semi-colon.

Reply:
Also it should be 
` pResponseHandle->pHeadersEnd = safeMalloc(sizeof(pResponseHandle)) + sizeof(_resHandle_t);`

Reply:
pHeadersEnd needs to point to the end of the same memory location as pHeaders and pHeadersCur

Reply:
Done.